### PR TITLE
Security documentation in the Cross cutting document

### PR DESCRIPTION
**Pull request by lwjohnst86**
@rqding can you take over finishing this? Basically review it, add anything that needs to be added (or removed).

### CODE REVIEW

File: design/cross-cutting-concepts.qmd [hunk 1]
```diff
@@ -1,15 +1,156 @@
...
+     +---------------------------------------+
+                      |
+     +---------------------------------------+
+     |            Database Management        |
+     |            System (DBMS)              |
+     +---------------------------------------+
+                      |
+     +---------------------------------------+
+     |            Database Security          |
+     +---------------------------------------+
```
Comment (lwjohnst86) [inline design/cross-cutting-concepts.qmd:52]: Can you make this as a Mermaid or PlantUML diagram? This would not render properly on the website.

**Comment by K-Beicher** [comment]
flowchart TD
    A[Authentication and \nAuthorization] --- B[API Security and Access]
    B --- C[Application Logic and \nData Validation]
    C --- D[Database Security section]




Med Venlig hilsen / Best regards,

Kristiane B Beicher
Databaseadministrator
***@***.******@***.***>

Steno Diabetes Center Aarhus


From: Luke W Johnston ***@***.***>
Date: Friday, 24 March 2023 at 13.33
To: steno-aarhus/seedcase-project ***@***.***>
Cc: Kristiane B Beicher ***@***.***>, Review requested ***@***.***>
Subject: Re: [steno-aarhus/seedcase-project] Security documentation in the Cross cutting document (PR #112)

@lwjohnst86 commented on this pull request.

________________________________

In design/cross-cutting-concepts.qmd<https://github.com/steno-aarhus/seedcase-project/pull/112#discussion_r1147522466>:

> +-   Security layer diagram

+

+     +---------------------------------------+

+     |          Authentication and           |

+     |             Authorization             |

+     +---------------------------------------+

+                      |

+     +---------------------------------------+

+     |         API Security and Access       |

+     +---------------------------------------+

+                      |

+     +---------------------------------------+

+     |       Application Logic and Data      |

+     |              Validation               |

+     +---------------------------------------+

+                      |

+     +---------------------------------------+

+     |            Database Management        |

+     |            System (DBMS)              |

+     +---------------------------------------+

+                      |

+     +---------------------------------------+

+     |            Database Security          |

+     +---------------------------------------+

Can you make this as a Mermaid or PlantUML diagram? This would not render properly on the website.

File: design/cross-cutting-concepts.qmd (hunk 1 above)
Comment (rqding) [reply design/cross-cutting-concepts.qmd:52]: Sure, do you want me to add a separate puml file for the diagram?

File: design/cross-cutting-concepts.qmd [hunk 2]
```diff
@@ -1,15 +1,148 @@
...
+    rectangle "API Security and Access" as api
+    rectangle "Application Logic and Data Validation" as logic
+    rectangle "Database Management System (DBMS)" as dbms
+    rectangle "Database Security" as db_security
+
+    auth --> api
+    api --> logic
+    logic --> dbms
+    dbms --> db_security
+    @enduml
```
Comment (lwjohnst86) [inline design/cross-cutting-concepts.qmd:44]: Sadly, plantuml can not be rendered within the document (unlike Mermaid, which can), so this needs to be in its own file.

**Comment by lwjohnst86** [comment]
Connected in #84

**Pull request merged**